test/ConfigDistributor.spec.js

```diff
@@ -24,177 +21,100 @@ describe('A DistributeConfig class', () => {
-        if (fs.existsSync(validDirName)) {
-          fs.rmdirSync(validDirName);
-        }
-        expect(fs.existsSync(validDirName)).to.be.false;
-      });
+          consolidatorMock = {
+            consolidate: () => {
+              return Promise.resolve(standAloneConfigFixture.input);
+            }
+          };
+          consolidateSpy = spy(consolidatorMock, 'consolidate');
+
+        fileSystemMock = {
+          writeFile: () => { return Promise.resolve; },
+          readFile: () => { return Promise.resolve; }
+        };
+        writeFileSpy = spy(fileSystemMock, 'writeFile');
+        readFileSpy = spy(fileSystemMock, 'readFile');
 
-      afterEach(() => {
-        fs.rmdirSync(validDirName);
-      });
+        paths = fixtures.paths;
+        expected = standAloneConfigFixture.expectedOutput;
 
-      it('returns a promise that will not be rejected', () => {
-        return expect(
-          ConfigDistributor.writeDirectory(validDirName)
-        ).to.not.be.rejected;
+        distributor = new ConfigDistributor(fileSystemMock, paths);
       });
 
-      it('writes the directory', () => {
-        return ConfigDistributor.writeDirectory(validDirName).then(() => {
-          return expect(fs.existsSync(validDirName)).to.be.true;
-        });
+      it('initiates config consolidation with the config paths supplied', () => {
+        return distributor.distribute(consolidatorMock, reportMock)
+          .then(() => {
+            expect(consolidateSpy.calledOnceWithExactly(fixtures.paths.config)).to.be.true;
+          });
       });
 
-    });
-
-  });
-
-  describe('writeFile static method', () => {
-
-    let filepath;
-    let data;
-
-    beforeEach(() => {
-      data = standAloneConfigFixture;
-      filepath = '/tempDir/fileFile.tmp';
-    });
-
-    context('when there is an error writing the file', () => {
-
-      let fileWriterRejectionMock;
-
-      beforeEach(() => {
-        fileWriterRejectionMock = () => {
-          return Promise.reject();
-        }
+      it('determines the correct data to distribute to the JavaScript layer', () => {
+        return distributor.distribute(consolidatorMock, reportMock)
+          .then(() => {
+            expect(writeFileSpy.withArgs(expected.js).calledOnce).to.be.true;
+          });
       });
 
-      it('returns a promise that will be rejected', () => {
-        return expect(
-          ConfigDistributor.writeFile(data, filepath, fileWriterRejectionMock)
-        ).to.be.rejected;
+      it('attempts to distribute the JavaScript layer to the correct path', () => {
+        const jsonPath = paths.output.jsonFileName;
+        const expectedDirectory = jsonPath.substring(0, jsonPath.lastIndexOf('/') + 1);
+        const expectedFilename = jsonPath.substring(jsonPath.lastIndexOf('/') + 1);
+
+        return distributor.distribute(consolidatorMock, reportMock)
+
+          .then(() => { return new Promise((resolve) => {
```

Comment:
this looks complex, can't the mocks return already resolved promises (e.g. `Promise.resolve()` results)?

Reply:
Fundamentally separating the allowance of the test doubles (they'll give back promises about files that will eventually resolve) from the test's verification (arguments that have been passed to the `Filesystem` methods)